### PR TITLE
Fashion is my profession - Procedural jumpsuit variants

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
I'm adding random colored jumpsuits to the game. they can be found in wardrobe, clothemate and sometimes in maint lockers

Each costume is randomly generated from parts painted in random colors, which gives a huge number of combinations of the appearance of such costumes.

## Why / Balance
I really like the idea of procedural content generation in such small things. It also gives players the feeling that they have something unique, as it is almost impossible to meet 2 identical suits in one shift.

## Technical details
now RandomSprite can work together with ClothingComponent

## Media
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
**Changelog**
:cl:
- add: Added random colored jumpsuits to the mixed wardrobes. Each costume is randomly generated from parts painted in random colors, which gives a huge number of combinations of the appearance of such costumes.

### CODE REVIEW

**Comment by UbaserB** [comment]
would it be possible to get an only-pants variation? Lots of people feel sad about the fact that the only ordinary pants are the boxing shorts.

**Comment by TheShuEd** [comment]
yeah, ok

**Comment by TheShuEd** [comment]
image
added

**Comment by lzk228** [comment]
oh almighty god of randomizing

**Comment by Hanzdegloker** [comment]
we need this to be worked into the loadout redux NOW

**Comment by Carmell0** [comment]
ED NEVER MISSES 🔥🔥🔥

**Comment by potato1234x** [comment]
best pr

**Comment by MACMAN2003** [comment]
there could be a sewing machine like the uniform printer but it allows you to mix and match your own jumpsuit

**Comment by TheShuEd** [comment]
image

**Comment by potato1234x** [comment]
> there could be a sewing machine like the uniform printer but it allows you to mix and match your own jumpsuit

the loom from tg

**Comment by TheShuEd** [comment]
image
new style

**Comment by brainfood1183** [comment]
> image new style

Why do some of the sprites have holes in the middle? its ok for the worn sprite but not for the icon sprite.

**Comment by TheShuEd** [comment]
The icon on the ground uses the same sprite as the one worn by the person.  And these particular shirts seem to be “on the naked body”

File: Resources/Textures/Clothing/Uniforms/procedural.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by TheShuEd for Space Station 14",
+  "size": {
```
Comment (metalgearsloth) [inline Resources/Textures/Clothing/Uniforms/procedural.rsi/meta.json:5]: Are these derived from anything or entirely original? They look similar to existing sprites.

**Review by metalgearsloth — APPROVED** [review]

File: Resources/Textures/Clothing/Uniforms/procedural.rsi/meta.json (hunk 1 above)
Comment (TheShuEd) [reply Resources/Textures/Clothing/Uniforms/procedural.rsi/meta.json:5]: I drew everything myself anyway, albeit based on the current sprites. I broke it down into layers, masks and such. Every sprite is 50% or more edited by me in one way or another

Comment (metalgearsloth) [reply Resources/Textures/Clothing/Uniforms/procedural.rsi/meta.json:5]: Should still credit the original ones if you used them as a base.

Comment (TheShuEd) [reply Resources/Textures/Clothing/Uniforms/procedural.rsi/meta.json]: Done

**Pull request merged**